### PR TITLE
Create POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by AkivaKn**
Task 6: Build endpoint using TDD including error
handling, and update endpoints.json accordingly

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -162,4 +162,62 @@ describe("/api/articles/:article_id/comments", () => {
...
+        .send(newComment)
+        .expect(201)
+        .then(({ body: { comment } }) => {
+          expect(comment.body).toBe(newComment.body);
+          expect(comment.author).toBe(newComment.username);
+          expect(comment.comment_id).toBeNumber();
+          expect(comment.votes).toBe(0);
+          expect(comment.created_at).toBeString();
+          expect(comment.article_id).toBe(2);
+        });
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:182]: object.containing might be a nice matcher here! :D

File: __tests__/app.test.js [hunk 2]
```diff
@@ -162,4 +162,62 @@ describe("/api/articles/:article_id/comments", () => {
...
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad request");
+        });
+    });
+  });
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:222]: Excellent! It might also be nice to add a test in when the body is an empty string? Although it depends on whether you would accept that into the database. One other test might be a username that does not exist in the database :)

File: db/controllers/comments-controller.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
-const { selectComments } = require("../models/comments-model")
+const { use } = require("../app");
```
Comment (nc-niamh-smith) [inline db/controllers/comments-controller.js:1]: what is this? :)

File: db/controllers/comments-controller.js [hunk 4]
```diff
@@ -7,4 +8,14 @@ exports.getComments = (req, res, next) => {
...
     .catch(next)
+}
+
+exports.postComments = (req, res, next) => {
+    const { article_id } = req.params;
+    const { username, body } = req.body;
+    return insertIntoComments(article_id,username,body)
+        .then((comment) => {
+        res.status(201).send({comment})
+        })
```
Comment (nc-niamh-smith) [inline db/controllers/comments-controller.js:19]: ^ lovely controller! Although I might edit the name to be 'postComment', as there's only one comment posted at a time!

**Review by nc-niamh-smith — COMMENTED** [review]
Wonderful completion of the ticket here Akiva! 🎉 

I've left a couple of comments about perhaps some extra tests you may want to consider. 😄
